src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -159,6 +163,19 @@ public static String getNodeSelectorFromNodeIdentifier(NodeInfo node) {
       node.getNodeId() + "\"]";
   }
 
+  public static void uploadPipelineFromFile(String filename) {
```

Comment:
Wouldn't it help if this method handled navigation to studio page, getting to import button, and naming the pipeline as well? It might be more re-usable that way. Something like:
```javascript
SeleniumDriver.openPage(Constants.BASE_STUDIO_URL + "pipelines");
    WaitHelper.waitForPageToLoad();
    ElementHelper.clickOnElement(locateElementById("resource-center-btn"));
    ElementHelper.clickOnElement(locateElementById("create-pipeline-link"));
    Assert.assertTrue(SeleniumDriver.getDriver().getCurrentUrl().contains("studio"));

    WaitHelper.waitForPageToLoad();
    WebElement uploadFile = SeleniumDriver.getDriver()
      .findElement(By.xpath("//*[@id='pipeline-import-config-link']"));

    File pipelineJSONFile = new File(Constants.FIXTURES_DIR + filename);
    String filePath = pipelineJSONFile.getAbsolutePath();
    uploadFile.sendKeys(filePath);

    String pipelineNameXPathSelector = "//div[contains(@class, 'PipelineName')]";
    SeleniumDriver.getWaitDriver().until(ExpectedConditions
                                           .stalenessOf(locateElementByLocator(By.xpath(pipelineNameXPathSelector))));
    ElementHelper.clickOnElement(locateElementByLocator(By.xpath(pipelineNameXPathSelector)));

    WebElement pipelineNameInput = locateElementById("pipeline-name-input");
    ElementHelper.clearElementValue(pipelineNameInput);
    ElementHelper.sendKeys(pipelineNameInput, pipelineName);
    pipelineNameInput.sendKeys(Keys.RETURN);
    ```

Reply:
Then it could be either saved as a draft, deployed, etc.

Reply:
This was extracted from `deployAndTestPipeline` which does those things (and deploys). It might make sense to extract other parts from `deployAndTestPipeline` like naming the pipeline, but I don't think those should be done here

Reply:
On a second thought, I prefer to leave `deployAndTestPipeline` as it is. If you only need part of it, just create another method